### PR TITLE
ISSUE-568 TeamMailbox/revokeAccess implementation

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/TeamMailboxRevokeAccessMethodContract.scala [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+         |}""".stripMargin)
+
+    assertThat(server.getProbe(classOf[TeamMailboxProbe]).listMembers(teamMailbox).asJava)
+      .isEmpty()
+  }
+
+  // TODO invalid mailbox name case should be fail
+  // TODO user revoke the same team mailbox 2 times (should it be idempotent?)
+  // TODO mixed case succeed and fail case
+  // TODO make sure delegatee can not access this API
```
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/TeamMailboxRevokeAccessMethodContract.scala:215]: Also add a test to unsure one no longer see in mailboxes/get a revoked team mailbox...

File: tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TeamMailbox.scala [hunk 2]
```diff
@@ -43,10 +44,56 @@ object TeamMailboxName {
...
+      throw new IllegalArgumentException("Missing '@' in mailbox FQDN")
+    }
+    val localPart = value.substring(0, atPosition)
+    val domainPart = value.substring(atPosition + 1)
+    TeamMailbox(domain = Domain.of(domainPart),
+      mailboxName = TeamMailboxName.fromString(localPart)
+        .fold((e: IllegalArgumentException) => throw e, (x: TeamMailboxName) => x))
+  }
+}
+
```
Comment (chibenwa) [inline tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TeamMailbox.scala:96]: IMO TeamMailboxFQDN is not needed and we should work only with TeamMailbox abstraction...

**Comment by quantranhong1999** [comment]
Ok seems the team intended to avoid `.` in team mailbox name before (I was not there that time so I don't know the rationale). I will drop/revert the https://github.com/linagora/tmail-backend/pull/577/commits/26a0fd59174136276e85669233c2cd8f190be6c8

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/json/TeamMailboxRevokeAccessSerializer.scala [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+object TeamMailboxRevokeAccessSerializer {
+  private implicit val unparsedTeamMailboxReads: Reads[UnparsedTeamMailbox] = {
+    case JsString(string) => JsSuccess(UnparsedTeamMailbox(string))
+    case _ => JsError("Team mailbox needs to be represented by a JsString")
+  }
+  private implicit val teamMailboxRevokeAccessRequestReads: Reads[TeamMailboxRevokeAccessRequest] = Json.reads[TeamMailboxRevokeAccessRequest]
+
+  private implicit val teamMailboxWrites: Writes[TeamMailbox] = teamMailbox => JsString(teamMailbox.asString())
+  private implicit val subscriptionMapSetErrorForCreationWrites: Writes[Map[UnparsedTeamMailbox, SetError]] =
```
Comment (vttranlina) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/json/TeamMailboxRevokeAccessSerializer.scala:17]: "subscription" ?

**Review by vttranlina — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by quantranhong1999** [comment]
Squashed.

**Comment by Arsnael** [comment]
Can you rebase it again? :)

**Pull request merged**